### PR TITLE
Add a draft of 0.6.0 ann post

### PR DESCRIPTION
**Pull request by solnic**

### CODE REVIEW

**Comment by solnic** [comment]
I'd love to push this tomorrow. Please take a look if you have time and let me know what you think /cc @elskwid @maetl @gotar @cflipse @splattael @aflatter

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  relation :users
+  register_as :create
+  result :one
+end
+
+rom = ROM.finalize.env
+
+create_user = rom.commands.users[:create]
+
+rom.command(:users).try { create_user.call(name: 'Jane') }
```
Comment (gotar) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: maybe will be good to show here the simples way to call it

```ruby
rom.command(:users).create.call(name: 'Jane')
``

without try block?

**Comment by gotar** [comment]
+1 :ship: it

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+## Class-based setup
+
+The biggest change in this release is the ability to define your own relation, mapper and command subclasses that ROM will instantiate for you and expose through its registry. The DSL-based setup is still available but using classes turned out to be a better approach in bigger contexts like web-applications.
+
+Here's a simple Relation subclass definition:
+
+``` ruby
+class Users < ROM::Relation[:memory]
+  def by_name(name)
+    where(name: name)
```
Comment (gotar) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: did `where` works in memory adapter?

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+``` ruby
+class UserPresenterMapper < ROM::Mapper
+  relation :users
+  register_as :presenter
+
+  model UserPresenter
+end
+
+user_presenter = rom.relation(:users).as(:entity, :presenter).by_name('Jane').first
```
Comment (gotar) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: you show `as` before maybe .`map_with` here? And some note about aliast

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+On top of that there's also an experimental partial-application feature where you can grab a reference to a relation with partially applied arguments:
+
+``` ruby
+users_by_name = rom.relation(:users).as(:entity).by_name
+
+users_by_name['Jane']
```
Comment (gotar) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: hmm you use `model` above then you will not have `hash` but `Object` here. For me it should be sth like:

```
users_by_name = rom.relation(:users).as(:entity).by_name

users_by_name.first.name # => Jane
```

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown (hunk 2 above)
Comment (maetl) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: Maybe `restrict`?

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 5]
```diff
@@ -0,0 +1,186 @@
+---
+title: ROM 0.6.0 Released
+date: 2015-03-22
+tags: release,announcement
+author: Piotr Solnica
+---
+
+We're happy to announce the final release of ROM 0.6.0! This version brings a lot of improvements and simplifies internal architecture of ROM. There are over 700 commits between 0.5.0 and this release which is a result of fantastic feedback that we've received from the early adopters and many contributions from new people.
```
Comment (maetl) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: A suggestion in terms of reading flow:

>  simplifies *the* internal architecture of ROM.

> received from ~~the~~ early adopters

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+This release includes updates of the following gems:
+
+* [rom 0.6.0](https://rubygems.org/gems/rom) - [changelog](https://github.com/rom-rb/rom/blob/master/CHANGELOG.md#v060-2015-03-22)
+* [rom-sql 0.4.0](https://rubygems.org/gems/rom-sql) - [changelog](https://github.com/rom-rb/rom-sql/blob/master/CHANGELOG.md#v040-2015-03-22)
+* [rom-rails 0.3.0](https://rubygems.org/gems/rom-rails) - [changelog](https://github.com/rom-rb/rom-rails/blob/master/CHANGELOG.md#v040-2015-03-22)
+
+Following new adapters have been released too:
```
Comment (maetl) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: > *The* following new adapters

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+class CreateUser < ROM::Commands::Create[:memory]
+  relation :users
+  register_as :create
+  result :one
+end
+
+rom = ROM.finalize.env
+
+create_user = rom.commands.users[:create]
```
Comment (maetl) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: `rom.commands.users[:create]`

Should this be `rom.commands.users(:create)`? Or does `[]` work as well?

Comment (solnic) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: Actually I should change it to `rom.command(:users).create`

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown (hunk 2 above)
Comment (solnic) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: thanks, that's a mistake, should be `restrict`

**Comment by solnic** [comment]
@gotar @maetl thanks :) I just pushed the fixes

File: source/blog/2015-03-22-rom-0-6-0-released.html.markdown (hunk 3 above)
Comment (solnic) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: I'm still not sure which one fits better :) mentioning alias would be too much here I think, it's not documentation, just some code snippets for people to look at, maybe I'm wrong though

Comment (maetl) [inline source/blog/2015-03-22-rom-0-6-0-released.html.markdown]: I agree that it’s a good idea to explain the use of both aliases, but for now, I’d recommend sticking to a single alias in all the blog post examples, to keep things consistent.

**Pull request closed without merging**